### PR TITLE
Upgrade MUI to V5 and fix a bunch of bad code

### LINKED ISSUE #225: Upgrade MUI to V5
Still haven't figured out a way with it yet. had issue with the styling engine. will try again sometime.

### PR DESCRIPTION
**Pull request by haihan-lin**
### Does this PR close any open issues?
Closes #225 

### Give a longer description of what this PR addresses and why it's needed

- Upgrade MUI (some of the "lab" features have been added to the official library).
- changed package manager from NPM to YARN
- New styling engine Emotion instead of Styled-Components
- Move dialog state management out of global state, into parts that actually trigger the dialog. 

### Provide pictures/videos of the behavior before and after these changes (optional)
Should all look the same with some minor aesthetic change

### TODO
- [ ] another thorough testing making sure no broken features

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -33,10 +32,10 @@
         "react-google-charts": "^4.0.0",
         "react-grid-layout": "^1.2.5",
         "react-idle-timer": "^4.6.4",
-        "react-scripts": "^4.0.3",
-        "styled-components": "^5.3.0",
+        "react-scripts": "^5.0.1",
         "typescript": "^4.3.4",
-        "use-deep-compare-effect": "^1.6.1"
+        "use-deep-compare-effect": "^1.6.1",
+        "yarn": "^1.22.19"
```
Comment (JackWilb) [inline frontend/package.json:38]: It's strange to have `yarn` here. We should probably just switch over to yarn fully, and remove package-lock.json, if that's what you'd like to use to manage the packages

**Review by JackWilb — APPROVED** [review]
I didn't dig too deep into the code changes, but the ones I looked at seemed to not be a problem. I have one note about yarn, but it shouldn't hold up merging

File: frontend/package.json (hunk 1 above)
Comment (haihan-lin) [reply frontend/package.json:38]: can you help me on making sure to clean up these? thank you very much

Comment (JackWilb) [reply frontend/package.json]: See 3186ff7

**Pull request merged**